src/Couchbase.Lite.Shared/API/Sync/ReplicatorConfiguration.cs

```diff
@@ -74,6 +74,12 @@ internal enum ReplicatorProgressLevel : int
         PerAttachment // >=2
     }
 
+    public class ReplicationFilter
```

Comment:
This is awkward and something I argued against when designing the API, so in cases like this we are allowed on .NET to just use functions (which is why there is a comment there saying `/* Use block or function instead if possible */`).  Revert this back to the way it was

Reply:
how do I pass isDeleted flag in the callback if use the way it was?

Reply:
It should be `Func<bool, Document, bool>` so that it is a function that takes a `Document` and a `bool`